### ISSUE TITLE
Access Violation SEGFAULT

### ISSUE THREAD

**Issue by davidsilveiro** [the-tcpdump-group/tcpdump#520]
Within ./print-ascii.c:90 it seems **s1 = *cp++** or **snprintf** below could be the culprit. Unsure on this one, needs more debugging.

First crash file takes a little long, just added second. 

tcpdump -r [crash file]

Slow Crash:
[crash.pcap.zip](https://github.com/the-tcpdump-group/tcpdump/files/287650/crash.pcap.zip)

Fast(er) Crash:
[faster-crash-poc.zip](https://github.com/the-tcpdump-group/tcpdump/files/287683/faster-crash-poc.zip)

**Comment by infrastation** [comment]
Is this on the latest master branch?

**Comment by davidsilveiro** [comment]
@infrastation I believe it is

**Comment by infrastation** [comment]
I cannot reproduce the segfault using the current master build of tcpdump and libpcap. In addition, line 90 of `print-ascii.c` is an empty line before the function `hex_and_ascii_print_with_offset()` as you can see [here](https://github.com/the-tcpdump-group/tcpdump/blob/master/print-ascii.c). Could you check once again?

**Comment by davidsilveiro** [comment]
Weird the reproduction isn't working out. Just chucking you a few more details on system build and package versions.

Distributor ID:	Ubuntu
Description:	Ubuntu 14.04.3 LTS
Release:	14.04
Codename:	trusty

tcpdump version 4.5.1
libpcap version 1.5.3


**Line 105 - 113**
```
nshorts = length / sizeof(u_short);
	i = 0;
	hsp = hexstuff; asp = asciistuff;
	while (--nshorts >= 0) {
		s1 = *cp++;
		s2 = *cp++;
		(void)snprintf(hsp, sizeof(hexstuff) - (hsp - hexstuff),
		    " %02x%02x", s1, s2);
		hsp += HEXDUMP_HEXSTUFF_PER_SHORT;
```

Fixed line translation, so the lines previously pointed out were 109 & 111.

@infrastation

**Comment by infrastation** [comment]
With tcpdump 4.5.1 and current libpcap the "slow" case does not reproduce (`tcpdump: unsupported pcap savefile version 12336.12336`), so it is most likely one of the libpcap bugs that have been fixed since libpcap 1.5.3 (1.7.5 is the latest stable release).

The "fast" case does reproduce on tcpdump 4.5.1 and current libpcap, subsequent tcpdump releases also had the same issue until release 4.7.0, which includes commit 4ac7226 made on 23 Oct 2014, which fixes this bug.

It should be more useful to test the current master branch or at the very least the latest stable release, which right now is 4.7.4, and which is available as a standard package in Ubuntu 16.04. Let us know if there is anything else to review in this request.

**Comment by davidsilveiro** [comment]
Thanks @infrastation for taking the time with this. I'll make sure to do just as you said, and I'd like to mention how fond I am of tcpdump. It's such a versatile application, keep up the great work!
